### DISCUSSION TITLE
What does "Deduplicating compilation" mean?

### DISCUSSION THREAD

**Discussion by davidmlw** [scalacenter/bloop#1263] [General]
moduleA depends on moduleB, and moduleB depends on moduleC.
```
bloop compile moduleC
bloop compile moduleB
bloop compile moduleA
```
in sequence, should work exactly as
```
bloop compile moduleA
```

While because the sequence was emitted from a flow tool, it might be too fast or parallel, bloop will complains "Deduplicating Compilation of moduleC".

It seems that "Deduplicating Compilation" didn't go wrong or multiple random classesDir.

**Comment by jvican** [comment]
@davidmlw Deduplicating compilation means that a compilation triggered by another client is reusing the compilation happening by another client. This can happen when Metals triggers a compilation and you're compiling/testing/running with the bloop CLI. It works as expected and means that bloop is making good use of resources.
